pkgs/applications/misc/st/default.nix

```diff
@@ -4,22 +4,20 @@
 with stdenv.lib;
 
 stdenv.mkDerivation rec {
-  name = "st-0.6";
+  name = "st";
+  version = "0.7";
   
   src = fetchurl {
-    url = "http://dl.suckless.org/st/${name}.tar.gz";
-    sha256 = "0avsfc1qp8zvshsfjwwrkvk411jlqy58z225bsdhjkl1qc40qcc5";
+    url = "http://dl.suckless.org/st/${name}-${version}.tar.gz";
+    sha1 = "358bca5bfc6ee85352dcd4f45f13f78474e5de42";
   };
-
-  inherit patches;
+  patches = [ ./st.patch ];
```

Comment:
This looks wrong, patches is an argument to the st function, either that argument should be removed, or the argument should be appended with st.patch.
```
 { stdenv, fetchurl, pkgconfig, writeText, libX11, ncurses, libXext, libXft, fontconfig
 , conf ? null, patches ? []}:
```